docs/README.md

```diff
@@ -485,6 +485,22 @@ the program's secret values to local storage, which means they can be read by
 program can start up immediately using cached data, even if the secrets server
 is not reachable when it launches.
 
+> [!WARNING]
+> When you enable a secrets cache for a program, new secret values may not
+> immediately become available even if the program is restarted. By design, if
+> a cached value is available at startup, the store does not wait for the
+> secrets service to respond before delivering the initial (cached) value.
+>
+> The store will see the new value (and update the cache) the next time it
+> successfully polls.  If the program only looks at the initial value of the
+> secret, however, it will not see the new value until it is restarted _after_
+> the next update.
+>
+> As a general rule, we recommend you _not_ enable a cache unless the program
+> cannot tolerate any delay at startup.  Otherwise, if you must use a cache, we
```

Comment:
Minor suggestion: I'd say something a bit stronger than "delay at startup" here, it's specifically being able to tolerate an outage of the secrets service, or being in the bootstrap path for same. The choice is less between "startup is fast" or "startup is slow", it's "does the thing start at all when the vault or tailscale are down"

Reply:
Agreed, let me see what I can do. I'm specifically calling this out because I keep seeing caches get enabled for things where they probably shouldn't.

Reply:
PTAL

Reply:
LGTM! And yeah, this was... I think awly's worry with having a cache, that it ends up being enabled for the wrong reasons.

Reply:
Indeed, and you and I discussed that at the time. I think the tradeoff was probably worth it, because it's better to enable it for the wrong reasons than to have someone continue to use ad-hoc secrets management because of that concern.